tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/service/websub/advanced/WebSubSecureHubTestCase.java

```diff
@@ -0,0 +1,112 @@
+ */
+
+package org.ballerinalang.test.service.websub.advanced;
+
+import org.ballerinalang.test.context.BServerInstance;
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.context.LogLeecher;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+import static org.ballerinalang.test.service.websub.WebSubTestUtils.CONTENT_TYPE_JSON;
+import static org.ballerinalang.test.service.websub.WebSubTestUtils.CONTENT_TYPE_XML;
+import static org.ballerinalang.test.service.websub.WebSubTestUtils.HUB_MODE_INTERNAL;
+import static org.ballerinalang.test.service.websub.WebSubTestUtils.HUB_MODE_REMOTE;
+import static org.ballerinalang.test.service.websub.WebSubTestUtils.PUBLISHER_NOTIFY_URL;
+import static org.ballerinalang.test.service.websub.WebSubTestUtils.PUBLISHER_NOTIFY_URL_THREE;
+import static org.ballerinalang.test.service.websub.WebSubTestUtils.requestUpdate;
+
+/**
+ * This class tests basic auth support for WebSub Hub.
+ * Tests:
+ * 1. Subscriber one is authenticated properly
+ * 2. Subscriber two fails the authentication due to incorrect password
+ * 3. Subscriber two fails the authorization due to incorrect scope
+ */
+@Test(groups = "websub-test")
+public class WebSubSecureHubTestCase extends WebSubAdvancedBaseTest {
+    private static final int LOG_LEECHER_TIMEOUT = 45000;
+    private BServerInstance webSubSubscriber;
+
+    private static final String INTENT_VERIFICATION_SUBSCRIBER_ONE_LOG = "ballerina: Intent Verification agreed - " +
+            "Mode [subscribe], Topic [http://one.persistence.topic.com], Lease Seconds [3600]";
+    private static final String INTENT_VERIFICATION_SUBSCRIBER_TWO_LOG =
+            "Subscription Request failed at Hub[https://localhost:9191/websub/hub], for Topic[http://two.persistence" +
+                    ".topic.com]: Error in request: Mode[subscribe] at Hub[https://localhost:9191/websub/hub] - " +
+                    "Authentication failure";
+    private static final String INTENT_VERIFICATION_SUBSCRIBER_THREE_LOG =
+            "Subscription Request failed at Hub[https://localhost:9191/websub/hub], for Topic[http://one.persistence" +
+                    ".topic.com]: Error in request: Mode[subscribe] at Hub[https://localhost:9191/websub/hub] - " +
+                    "Authorization failure ";
+    private static final String INTENT_VERIFICATION_SUBSCRIBER_FOUR_LOG = "ballerina: Intent Verification agreed - " +
+            "Mode [subscribe], Topic [http://one.websub.topic.com], Lease Seconds [1200]";
+    private static final String INTERNAL_HUB_NOTIFICATION_SUBSCRIBER_ONE_LOG =
+            "WebSub Notification Received by One: {\"mode\":\"internal\", \"content_type\":\"json\"}";
+    private static final String INTERNAL_HUB_NOTIFICATION_SUBSCRIBER_TWO_LOG =
+            "WebSub Notification Received by Four: {\"mode\":\"remote\", \"content_type\":\"xml\"}";
```

Comment:
should we maybe test this based on a different payload? 

`{\"mode\":\"remote\", \"content_type\":\"xml\"}` was originally meant to remotely publish an xml payload to the hub.